app/controllers/data_exports_controller.rb

```diff
@@ -0,0 +1,17 @@
+class DataExportsController < ApplicationController
+  respond_to :csv
+
+  def csv
+    type = params[:type]
+
+    respond_to do |format|
+      format.csv { send_data data(type), filename: "#{type}-#{Time.zone.today}.csv" }
```

Comment:
The type gets white-listed in the export itself, but here you could inject Evil Things into the filename I think.